src/main/reactjs/src/redux/sagas/clerkUser.ts

```diff
@@ -0,0 +1,47 @@
+import { takeLatest, call, put } from '@redux-saga/core/effects';
+import { AxiosResponse } from 'axios';
+
+import axiosInstance from 'configs/axios';
+import { ExternalAPIEndpoints } from 'enums/external';
+import { HTTPStatusCode } from 'enums/app';
+import { ClerkUserAPIResponse } from 'interfaces/clerkUser';
+import { setClerkUser } from 'redux/actions/clerkUser';
+import {
+  CLERK_USER_ERROR,
+  CLERK_USER_LOAD,
+  CLERK_USER_MOCK_LOAD,
+} from 'redux/actionTypes/clerkUser';
+import { APIResponseStatus } from 'enums/api';
+
+export function* fetchClerkUser() {
+  try {
+    const response: AxiosResponse<ClerkUserAPIResponse> = yield call(
+      axiosInstance.get,
+      ExternalAPIEndpoints.ClerkUser
+    );
+
+    if (response.status === HTTPStatusCode.Ok) {
+      yield put(setClerkUser(response.data));
+    }
```

Comment:
Pitäisikö tässä poistaa kuitenkin if-clause ja yieldata aina tuo put? Lähinnä mietin sitä, että tulevaisuudessa tätä koodia lukiessa voi ihmetellä miksei muita ei-virheellisiä paluuarvoja käsitelty.

Reply:
Siis tämä tulee muuttumaan, kun otetaan meidän oma rajapinta clerkUseria varten, mutta tällä hetkellä saa jatkaa ainoastaan kun status on ok.

Reply:
Jos käyttäjä ei ole kirjautunut niin vastauksena tulee 30X koodeja tällä hetkellä, koska käytämme clerkME rajapintaa, mutta vaihdetaan se ja käytetään omaa rajapintaa taskissa OPHAKTKEH-195

Reply:
Jos kuitenkin tämän kirjautumis-flow'n kannalta muut kuin 200 on oleellisesti virheellisiä, niin silloin voisi else-haarassa myös lähettää error-actionin.

Reply:
Kyllä, mutta kuten sanoin tämä tulee muuttumaan pian, lisätään kuitenkin elseen nyt error action.

Reply:
Ajatukseni tämän osalta siis olisi että jos kutsuttu endpoint vastaa jotain mitä ei odotettu (= 200), niin silloin kirjautumis-flow'n ei kuitenkaan pitäisi mennä "hiljaa" läpi.

Reply:
Toisaalta se voi unohtua silloin :) Tosiaan kun tulee meidän oma rajapinta if poistuu kokonaan.

Reply:
Jatkaisin kyllä tällä ja tekisin noita muutoksia OPHAKTKEH-195:n yhteydessä.

Reply:
Tehdään näin.